### PR TITLE
Fix some phpdoc

### PR DESCRIPTION
**Pull request by alamirault**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -218,7 +218,7 @@ private static function doRemove(array $files, bool $isRecursive): void
     public function chmod($files, int $mode, int $umask = 0000, bool $recursive = false)
     {
         foreach ($this->toIterable($files) as $file) {
-            if ((\PHP_VERSION_ID < 80000 || \is_int($mode)) && !self::box('chmod', $file, $mode & ~$umask)) {
```
Comment (alamirault) [inline src/Symfony/Component/Filesystem/Filesystem.php]: `\is_int($mode)` it always true

Comment (OskarStark) [inline src/Symfony/Component/Filesystem/Filesystem.php]: Why removing the php check?

Comment (alamirault) [reply src/Symfony/Component/Filesystem/Filesystem.php]: `$mode` is type hinted `int $mode`, so this condition can't be false.

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Filesystem/Filesystem.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Component/Filesystem/Filesystem.php]: Although you are correct, I think we don't do this kind of change on LTS branches anymore.

Comment (alamirault) [reply src/Symfony/Component/Filesystem/Filesystem.php]: I wasn't sure also :smile: 

I rollbacked this and created a PR on 6.3 branch https://github.com/symfony/symfony/pull/49341

File: src/Symfony/Component/Finder/Comparator/NumberComparator.php [hunk 2]
```diff
@@ -35,7 +35,7 @@
 class NumberComparator extends Comparator
 {
     /**
-     * @param string|int $test A comparison string or an integer
+     * @param string|null $test A comparison string or an integer
```
Comment (fabpot) [inline src/Symfony/Component/Finder/Comparator/NumberComparator.php]: The description should be updated then?

**Review by fabpot — CHANGES_REQUESTED** [review]

**Review by yceruto — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alamirault.

**Pull request merged**